app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -440,40 +448,44 @@ fun CommentItem(
               }
         }
 
-        // Toggle button to show/hide the reply input field
-        Button(
-            onClick = { showReplyField = !showReplyField },
-            modifier =
-                Modifier.padding(top = 4.dp)
-                    .testTag("${if (showReplyField) "Cancel" else "Reply"}Button_${comment.uid}")) {
-              Text(if (showReplyField) "Cancel" else "Reply")
-            }
-      }
+        if (allowReplies) {
+          // Toggle button to show/hide the reply input field
+          Button(
+              onClick = { showReplyField = !showReplyField },
+              modifier =
+                  Modifier.padding(top = 4.dp)
+                      .testTag(
+                          "${if (showReplyField) "Cancel" else "Reply"}Button_${comment.uid}")) {
+                Text(if (showReplyField) "Cancel" else "Reply")
+              }
+        }
 
-      // Conditionally show the reply input field if the user is logged in
-      if (showReplyField) {
-        OutlinedTextField(
-            value = replyText,
-            onValueChange = { replyText = it },
-            label = { Text("Reply") },
-            modifier = Modifier.fillMaxWidth().testTag("replyInputField_${comment.uid}"))
+        // Conditionally show the reply input field if the user is logged in
+        if (showReplyField) {
+          OutlinedTextField(
+              value = replyText,
+              onValueChange = { replyText = it },
+              label = { Text("Reply") },
+              modifier = Modifier.fillMaxWidth().testTag("replyInputField_${comment.uid}"))
 
-        Button(
-            onClick = {
-              onReplyComment(replyText, comment)
-              replyText = ""
-              showReplyField = false
-            },
-            modifier = Modifier.padding(top = 4.dp).testTag("postReplyButton_${comment.uid}")) {
-              Text("Post Reply")
-            }
+          Button(
+              onClick = {
+                onReplyComment(replyText, comment)
+                replyText = ""
+                showReplyField = false
+              },
+              modifier = Modifier.padding(top = 4.dp).testTag("postReplyButton_${comment.uid}")) {
+                Text("Post Reply")
+              }
+        }
```

Comment:
The implementation for handling replies using an OutlinedTextField and a Button is well-structured. It provides a clear and straightforward mechanism for users to enter and submit their replies.